### PR TITLE
[onert-micro] Extend OMRuntimeModule

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This pr extends OMRuntimeModule to support wof files. Also this pr renames executeKernel to runForward for training purposes.

for issue https://github.com/Samsung/ONE/issues/12873
from draft: https://github.com/Samsung/ONE/pull/13107

Note: Should be rebased after #13213 will be merged

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

File: onert-micro/onert-micro/src/execute/CMakeLists.txt [hunk 1]
```diff
@@ -67,3 +67,4 @@ GTest_AddTest(${OM_EXECUTE_LIB}_kernels_test ${TEST_SOURCES})
 target_include_directories(${OM_EXECUTE_LIB}_kernels_test PUBLIC "${OM_INCLUDE_DIR}")
 target_link_libraries(${OM_EXECUTE_LIB}_kernels_test ${OM_INTERPRETER_LIB})
 target_link_libraries(${OM_EXECUTE_LIB}_kernels_test onert_micro_coverage)
+target_link_libraries(${OM_EXECUTE_LIB} PUBLIC onert_micro_coverage)
```
Comment (BalyshevArtem) [inline onert-micro/onert-micro/src/execute/CMakeLists.txt:70]: Fixes to obtain coverage

File: onert-micro/onert-micro/src/core/OMRuntimeModule.cpp [hunk 2]
```diff
@@ -133,7 +138,7 @@ OMStatus OMRuntimeModule::run()
   execute::OMExecuteArgs execute_args = {main_graph.getRuntimeStorage(),
                                          main_graph.getRuntimeContext(), 0, *this};
 
-  status = execute::OMKernelExecute::executeKernel(execute_args, main_graph.getRuntimeAllocator());
+  status = execute::OMKernelExecute::runForward(execute_args, main_graph.getRuntimeAllocator());
```
Comment (BalyshevArtem) [inline onert-micro/onert-micro/src/core/OMRuntimeModule.cpp:141]: Just to rename to separate forward and backward execution

**Comment by BalyshevArtem** [comment]
Rebased

**Review by Torrero — APPROVED** [review]
LGTM

**Review by ljwoo94 — DISMISSED** [review]
LGTM

**Comment by ljwoo94** [comment]
```
/home/runner/work/ONE/ONE/onert-micro/onert-micro/include/core/OMRuntimeContext.h:63: undefined reference to `onert_micro::core::reader::OMWeightOnlyFormatReader::validate(onert_micro::core::reader::OMCircleReader*)'
/usr/bin/ld: ../core/libonert_micro_core.a(OMRuntimeContext.cpp.o): in function `onert_micro::core::OMRuntimeContext::getConstDataByTensorIndex(unsigned char**, unsigned short)':
/home/runner/work/ONE/ONE/onert-micro/onert-micro/src/core/OMRuntimeContext.cpp:64: undefined reference to `onert_micro::core::reader::OMWeightOnlyFormatReader::buffer(unsigned int)'
collect2: error: ld returned 1 exit status
```

**Comment by ljwoo94** [comment]
Need to add `reader/OMWeightOnlyFormatReader.cpp` in `src/core/CmakeLists.txt` to resolve above compile error.

**Comment by BalyshevArtem** [comment]
> Need to add `reader/OMWeightOnlyFormatReader.cpp` in `src/core/CmakeLists.txt` to resolve above compile error.

Done

**Review by ljwoo94 — APPROVED** [review]

**Pull request merged**